src/storage/remote/protocol.c

```diff
@@ -143,18 +142,22 @@ storageRemoteInfoWrite(ProtocolServer *server, const StorageInfo *info)
     FUNCTION_TEST_END();
 
     storageRemoteInfoWriteType(server, info->type);
-    protocolServerWriteLine(server, jsonFromUInt(info->userId));
-    protocolServerWriteLine(server, jsonFromStr(info->user));
-    protocolServerWriteLine(server, jsonFromUInt(info->groupId));
-    protocolServerWriteLine(server, jsonFromStr(info->group));
-    protocolServerWriteLine(server, jsonFromUInt(info->mode));
     protocolServerWriteLine(server, jsonFromInt64(info->timeModified));
 
     if (info->type == storageTypeFile)
         protocolServerWriteLine(server, jsonFromUInt64(info->size));
 
-    if (info->type == storageTypeLink)
-        protocolServerWriteLine(server, jsonFromStr(info->linkDestination));
+    if (info->level >= storageInfoLevelDetail)
+    {
+        protocolServerWriteLine(server, jsonFromUInt(info->userId));
+        protocolServerWriteLine(server, jsonFromStr(info->user));
+        protocolServerWriteLine(server, jsonFromUInt(info->groupId));
+        protocolServerWriteLine(server, jsonFromStr(info->group));
+        protocolServerWriteLine(server, jsonFromUInt(info->mode));
+
+        if (info->type == storageTypeLink)
+            protocolServerWriteLine(server, jsonFromStr(info->linkDestination));
```

Comment:
Can you help me to understand why storageTypeLink is only written to protocol server if the level >= storageLevelDetail. I know there is still a "default" level and why wouldn't storageInfoLevelExists not write out link?

Reply:
Links require Posix and Posix requires storageInfoLevelDetail so they kind of got tied together. No reason this can't be split out, though, if you think it will be clearer.

Reply:
Actually, this is specified in the StorageInfo type as only being returned at level detail. I'm OK wit leaving it that way.